### PR TITLE
Feat: Rollback support added for devtron charts.

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

User not able to go back to previous values as we don't have build history ye. added API support where user can deploy re-deploy to a specific older configuration with same chart version or other chart version.

User now able to rollback to specific older version of deployment history.

Fixes # (issue) :[1298](https://github.com/devtron-labs/devtron/issues/1298)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Create a build history of chart version, Deploy multiple times
- [x] Rollback to older version.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go [hunk 1]
```diff
@@ -297,3 +303,106 @@ func (impl AppStoreDeploymentFullModeServiceImpl) createInArgo(chartGitAttribute
 
 	return nil
 }
+
+func (impl AppStoreDeploymentFullModeServiceImpl) UpdateValuesYaml(installAppVersionRequest *appStoreBean.InstallAppVersionDTO) (*appStoreBean.InstallAppVersionDTO, error) {
+	acdAppName := fmt.Sprintf("%s-%s", installAppVersionRequest.AppName, installAppVersionRequest.EnvironmentName)
+	gitOpsRepoName := installAppVersionRequest.GitOpsRepoName
+	if len(gitOpsRepoName) == 0 {
```
Comment (pghildiyal) [inline pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go:310]: i think we are using similar code to fetch gitops repo at other places also.. should extact it out as function

File: pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go [hunk 2]
```diff
@@ -297,3 +303,106 @@ func (impl AppStoreDeploymentFullModeServiceImpl) createInArgo(chartGitAttribute
...
+		FileName:       appStoreBean.REQUIREMENTS_YAML_FILE,
+		FileContent:    string(requirementDependenciesByte),
+		ChartName:      appStoreAppVersion.AppStore.Name,
+		ChartLocation:  acdAppName,
+		ChartRepoName:  installAppVersionRequest.GitOpsRepoName,
+		ReleaseMessage: fmt.Sprintf("release-%d-env-%d ", appStoreAppVersion.Id, installAppVersionRequest.EnvironmentId),
+	}
+	gitOpsConfigBitbucket, err := impl.gitOpsRepository.GetGitOpsConfigByProvider(util.BITBUCKET_PROVIDER)
+	if err != nil {
+		if err == pg.ErrNoRows {
```
Comment (pghildiyal) [inline pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go:395]: this code should be inside GetGitOpsConfigByProvider

File: pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go (hunk 1 above)
Comment (vikramdevtron) [reply pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go]: done

File: pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go:403]: GetGitOpsConfigByProvider has only its global config services

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**